File: javascript/Menu_Driven_Dynamic.js
function outputArray(a) {
    console.log("Array elements: " + a.join(' '));
}

function linearSearch(a) {
    let n = parseInt(prompt("Enter n: "));
    let flag = false;
    for (let i = 0; i < a.length; i++) {
        if (a[i] === n) {
            console.log("Entered element " + a[i] + " found at index " + i);
            flag = true;
        }
    }
    if (!flag) {
        console.log("Entered element " + n + " not found in the array");
    }
}

function reverseArray(a) {
    a.reverse();
    console.log("Reversed array: " + a.join(' '));
}

function updateArray(a) {
    for (let i = 0; i < a.length; i++) {
        if (i % 2 === 0) {
            a[i] *= 2;
        } else {
            a[i] += 5;
        }
    }
    console.log("Updated array: " + a.join(' '));
}

function insertElement(a) {
    let index = parseInt(prompt("Enter index you want to insert element in: "));
    let element = parseInt(prompt("Enter element you want to insert: "));
    a.splice(index, 0, element);
    console.log("Array after insertion:");
    outputArray(a);
}

function deleteElement(a) {
    let index = parseInt(prompt("Enter index to be deleted: "));
    if (index < 0 || index >= a.length) {
        console.log("Invalid index");
        return;
    }
    a.splice(index, 1);
    console.log("Array after deletion:");
    outputArray(a);
}

function main() {
    let size = parseInt(prompt("Enter size of array: "));
    let a = [];
    console.log("Enter elements in the array:");
    for (let i = 0; i < size; i++) {
        let element = parseInt(prompt("Enter element: "));
        a.push(element);
    }
    outputArray(a);

    let choice;
    do {
        console.log("\nMenu:");
        console.log("1. Linear Search");
        console.log("2. Reverse Array");
        console.log("3. Update Array");
        console.log("4. Insert Element");
        console.log("5. Delete Element");
        console.log("6. Exit");

        choice = parseInt(prompt("Enter your choice: "));
        switch (choice) {
            case 1:
                linearSearch(a);
                break;
            case 2:
                reverseArray(a);
                break;
            case 3:
                updateArray(a);
                break;
            case 4:
                insertElement(a);
                break;
            case 5:
                deleteElement(a);
                break;
            case 6:
                console.log("Exiting program.");
                break;
            default:
                console.log("Invalid choice. Please enter a valid option.");
                break;
        }
    } while (choice !== 6);
}

main();

File: PYTHON/Menu_Driven_Dynamic.py
def output_array(a):
    print("Array elements:", *a)

def linear_search(a):
    n = int(input("Enter n: "))
    flag = False
    for i, item in enumerate(a):
        if item == n:
            print(f"Entered element {item} found at index {i}")
            flag = True
    if not flag:
        print(f"Entered element {n} not found in the array")

def reverse_array(a):
    a.reverse()
    print("Reversed array:", *a)

def update_array(a):
    for i in range(len(a)):
        if i % 2 == 0:
            a[i] *= 2
        else:
            a[i] += 5
    print("Updated array:", *a)

def insert_element(a):
    index = int(input("Enter index you want to insert element in: "))
    element = int(input("Enter element you want to insert: "))
    a.insert(index, element)
    print("Array after insertion:")
    output_array(a)

def delete_element(a):
    index = int(input("Enter index to be deleted: "))
    if index < 0 or index >= len(a):
        print("Invalid index")
        return
    del a[index]
    print("Array after deletion:")
    output_array(a)

def main():
    size = int(input("Enter size of array: "))
    a = []
    print("Enter elements in the array:")
    for _ in range(size):
        element = int(input("Enter element: "))
        a.append(element)
    output_array(a)

    while True:
        print("\nMenu:")
        print("1. Linear Search")
        print("2. Reverse Array")
        print("3. Update Array")
        print("4. Insert Element")
        print("5. Delete Element")
        print("6. Exit")

        choice = int(input("Enter your choice: "))
        if choice == 1:
            linear_search(a)
        elif choice == 2:
            reverse_array(a)
        elif choice == 3:
            update_array(a)
        elif choice == 4:
            insert_element(a)
        elif choice == 5:
            delete_element(a)
        elif choice == 6:
            print("Exiting program.")
            break
        else:
            print("Invalid choice. Please enter a valid option.")

if __name__ == "__main__":
    main()
